pulp_rpm/app/tasks/publishing.py

```diff
@@ -95,21 +92,46 @@ def prepare_metadata_files(self, contents, folder=None):
 
         return repomdrecords
 
-    def get_packages(self, contents):
+    def publish_artifacts(self, content):
         """
-        Get packages from content.
+        Publish artifacts.
 
         Args:
-            contents (pulpcore.plugin.models.Content): A list of contents.
+            content (pulpcore.plugin.models.Content): content set.
 
-        Returns:
-            packages (pulp_rpm.models.Package): A list of packages.
+        """
+        published_artifacts = []
+        for content_artifact in ContentArtifact.objects.filter(content__in=content).iterator():
+            published_artifacts.append(PublishedArtifact(
+                relative_path=content_artifact.relative_path,
+                publication=self.publication,
+                content_artifact=content_artifact)
+            )
 
+        PublishedArtifact.objects.bulk_create(published_artifacts, batch_size=2000)
+
+    def handle_sub_repo(self, lst, identifier):
```

Comment:
And what does it do? It only publishes artifacts, not metadata, correct?
how about `publish_sub_repo_artifacts`?

Reply:
I think it would be clearer for future code readers if the names are more descriptive.
How about `lst`->`sub_repos` and `identifier` -> `sub_repo_type`?

Reply:
it not only publish artifacts, it also makes sub repo content available to be used on `create_repomd_xml`

Reply:
identifier would be `addon_id` or `variant_id`, I don't know if `sub_repo_type` suits well